src/st/cen/hd_e_room_fg.c

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
```

Comment:
This is a unique version of `e_room_fg` and is not compatible with WRP.

Originally I had this conditionally compiling for `VERSION_HD`, but I would need to add a stage condition as well. I decided to move it to a separate file instead, but can move it back if that's preferred.

Besides the data differences, the condition and operations on lines 27-29 differ from the implementation shared.

Reply:
I changed this to follow the model of https://github.com/Xeeynamo/sotn-decomp/pull/1421 where the differenced code is local to the stage C file.